Index: backports-20200628-4.4.60-9a94b73e75/drivers/net/wireless/ath/ath11k/debug.c
===================================================================
--- backports-20200628-4.4.60-9a94b73e75.orig/drivers/net/wireless/ath/ath11k/debug.c
+++ backports-20200628-4.4.60-9a94b73e75/drivers/net/wireless/ath/ath11k/debug.c
@@ -1206,7 +1206,7 @@ static const struct file_operations fops
 };
 
 static ssize_t ath11k_debug_write_fw_recovery(struct file *file,
-                                              char __user *user_buf,
+                                              const char __user *user_buf,
                                               size_t count, loff_t *ppos)
 {
        struct ath11k_base *ab = file->private_data;
Index: backports-20200628-4.4.60-9a94b73e75/drivers/net/wireless/ath/ath11k/core.h
===================================================================
--- backports-20200628-4.4.60-9a94b73e75.orig/drivers/net/wireless/ath/ath11k/core.h
+++ backports-20200628-4.4.60-9a94b73e75/drivers/net/wireless/ath/ath11k/core.h
@@ -410,7 +410,7 @@ struct ath11k_sta {
 
 #define ATH11K_MIN_5G_FREQ 4150
 #define ATH11K_MIN_6G_FREQ 5925
-#define ATH11K_MAX_6G_FREQ 7115
+#define ATH11K_MAX_6G_FREQ 7105
 #define ATH11K_NUM_CHANS 101
 #define ATH11K_MAX_5G_CHAN 173
 
